app/models/invitation.rb

```diff
@@ -0,0 +1,22 @@
+class Invitation < ActiveRecord::Base
+
+	belongs_to :sender, :class_name => 'User'
+	has_one :recipient, :class_name => 'User'
+
+	validates :recipient_email, 
+			presence: true, 
+			format: { with: /\A([^@\s]+)@((?:[-a-z0-9]+\.)+[a-z]{2,})\z/i, on: :create }, 
```

Comment:
We already use Devise, which has nice helper for validating email. Please google.